src/query/functions/src/scalars/bitmap.rs

```diff
@@ -62,19 +67,26 @@ pub fn register(registry: &mut FunctionRegistry) {
         }),
     );
 
-    registry.register_passthrough_nullable_1_arg::<ArrayType<UInt64Type>, BitmapType, _, _>(
-        "build_bitmap",
-        |_, _| FunctionDomain::Full,
-        vectorize_with_builder_1_arg::<ArrayType<UInt64Type>, BitmapType>(|arg, builder, _ctx| {
-            let mut rb = RoaringTreemap::new();
-            for a in arg.iter() {
-                rb.insert(*a);
-            }
+    for num_type in ALL_UNSIGNED_INTEGER_TYPES {
+        with_integer_mapped_type!(|NUM_TYPE| match num_type {
+            NumberDataType::NUM_TYPE => {
+                registry.register_passthrough_nullable_1_arg::<ArrayType<NullableType<NumberType<NUM_TYPE>>>, BitmapType, _, _>(
+                    "build_bitmap",
+                    |_, _| FunctionDomain::Full,
+                    vectorize_with_builder_1_arg::<ArrayType<NullableType<NumberType<NUM_TYPE>>>, BitmapType>(|arg, builder, _ctx| {
+                        let mut rb = RoaringTreemap::new();
+                        arg.iter().filter(|a| a.is_some()).for_each(|a| {
+                            rb.insert(a.unwrap() as u64);
```

Comment:
If the value is a negative number, directly casting to u64 may give the wrong result.

Reply:
It looks like it only supports the unsigned integer type, can we also support the signed integer type and report an error when encountering a negative number?

Reply:
sounds good, let's do it.